### PR TITLE
Fixes before 0.2.0 release

### PR DESCRIPTION
**Pull request by malcolmholmes**
see comments inline to explain these changes.

### CODE REVIEW

File: pkg/grafana/dashboards.go [hunk 1]
```diff
@@ -96,6 +96,9 @@ func postDashboard(resource grizzly.Resource) error {
 		return err
 	}
 
+	if !resource.HasMetadata("folder") {
+		return fmt.Errorf("Resource %s requires a folder within its metadata", resource.GetMetadata("name"))
+	}
```
Comment (malcolmholmes) [inline pkg/grafana/dashboards.go:101]: complain if there is no `folder` metadata within a dashboard.

File: pkg/grafana/rules.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func getRemoteRuleGroup(uid string) (*grizzly.Resource, error) {
 						"rules": group.Rules,
 					}
 					handler := RuleHandler{}
-					resource := grizzly.NewResource(handler.APIVersion(), handler.Kind(), uid, spec)
+					resource := grizzly.NewResource(handler.APIVersion(), handler.Kind(), group.Name, spec)
```
Comment (malcolmholmes) [inline pkg/grafana/rules.go:55]: `uid` is `$namespace.$name`, so this misreports the name portion.

File: pkg/grizzly/parsing.go [hunk 3]
```diff
@@ -133,7 +133,11 @@ func ParseJsonnet(registry Registry, jsonnetFile string, opts Opts) (Resources,
 		if err != nil {
 			return nil, err
 		}
-		resources = append(resources, parsedResources...)
+		for _, parsedResource := range parsedResources {
+			if parsedResource.MatchesTarget(opts.Targets) {
+				resources = append(resources, parsedResource)
+			}
+		}
```
Comment (malcolmholmes) [inline pkg/grizzly/parsing.go:140]: somehow we lost our `-t` target handling functionality when parsing with Jsonnet.

File: pkg/grafana/utils.go [hunk 4]
```diff
@@ -9,6 +11,9 @@ const generalFolderId = 0
 func extractFolderUID(d DashboardWrapper) string {
 	folderUid := d.Meta.FolderUID
 	if folderUid == "" {
+		if d.Meta.FolderURL == "" {
+			return "general"
+		}
```
Comment (malcolmholmes) [inline pkg/grafana/utils.go:16]: Grafana can come back with a blank folder URL for the general folder.

File: pkg/grafana/utils_test.go [hunk 5]
```diff
@@ -27,10 +28,12 @@ func TestExtractFolderUID(t *testing.T) {
 		}
 		getFolderById = func(folderId int64) (Folder, error) {
 			return Folder{
-				"uid": "12345",
+				"uid":       "12345",
+				"folderURL": "http://localhost:3000/blah",
 			}, nil
 		}
 		uid := extractFolderUID(dashboardWrapper)
 		require.Equal(t, "12345", uid)
```
Comment (dsotirakis) [inline pkg/grafana/utils_test.go:36]: Need to also check for folderUrl here?

**Comment by malcolmholmes** [comment]
This work has all been done in other PRs.

**Pull request closed without merging**